core/templates/pages/contributor-dashboard-page/modal-templates/translation-modal.component.ts

```diff
@@ -0,0 +1,180 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Component for the translation modal.
+ */
+
+import { Component, Input } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+
+import { AlertsService } from 'services/alerts.service';
+import { CkEditorCopyContentService } from 'components/ck-editor-helpers/ck-editor-copy-content-service';
+import { ContextService } from 'services/context.service';
+import { ImageLocalStorageService } from 'services/image-local-storage.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { TranslateTextService } from 'pages/contributor-dashboard-page/services/translate-text.service';
+import { TranslationLanguageService } from 'pages/exploration-editor-page/translation-tab/services/translation-language.service';
+import { AppConstants } from 'app.constants';
+
+class UiConfig {
+  'hide_complex_extensions': boolean;
+  'startupFocusEnabled'?: boolean;
+  'language'?: string;
+  'languageDirection'?: string;
+}
+export class TranslationOpportunityDict {
+  id: string;
+  heading: string;
+  subheading: string;
+  progressPercentage: string;
+  actionButtonTitle: string;
+}
+export interface HTMLSchema {
+    'type': string;
+    'ui_config': UiConfig;
+}
+
+@Component({
+  selector: 'translation-modal',
+  templateUrl: './translation-modal.component.html',
+  styleUrls: [],
+  providers: []
+})
+export class TranslationModalContent {
```

Comment:
Why are we creating a component for a modal?

Reply:
because NgbModal needs a component as its content unlike $uibModal